TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,646 @@
+        }
+    }
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//   GYRO
+    public boolean gyroTurn(Direction direction, double power, float degrees){
+        switch(direction) {
+            case TURNLEFT:
+                if (goalDegrees == -1) {
+                    goalDegrees = (this.getYaw() - degrees);
+                    if (goalDegrees < 0) {
+                        goalDegrees = (goalDegrees + 360);
+                    }
+                }
+                if (!(this.getYaw() > (goalDegrees - gyroRange) && this.getYaw() < (goalDegrees + gyroRange))) {
+                    Drive(direction, power);
+                    telemetry.addData("Goal Degrees", goalDegrees);
+                    telemetry.addData("Current Degrees", getYaw());
+                    return false;
+                } else {
+                    this.stop();
+                    goalDegrees = -1;
+                    return true;
+                }
+            case TURNRIGHT:
+                if (goalDegrees == -1) {
+                    goalDegrees = (this.getYaw() + degrees);
+                    if (goalDegrees > 360) {
+                        goalDegrees = (goalDegrees - 360);
+                    }
+                }
+                if (!(this.getYaw() > (goalDegrees - gyroRange) && this.getYaw() < (goalDegrees + gyroRange))) {
+                    Drive(direction, power);
+                    telemetry.addData("Goal Degrees", goalDegrees);
+                    telemetry.addData("Current Degrees", getYaw());
+                    return false;
+                } else {
+                    this.stop();
+                    goalDegrees = -1;
+                    return true;
+                }
+        }
+        return false;
+    }
+
+    public float getYaw(){
+        if(this.gyro.getYaw() < 0) {
+            return (360 + this.gyro.getYaw());
+        } else {
+            return this.gyro.getYaw();
+        }
+    }
+
+    public void experimentalDrive(double x, double y, double z){
+        /*
+        Guide to motor Powers:
+        Left Front: - (y + x + Z)
+        Right Front: y - x - z
+        Left Rear: y + x - z
+        Right Rear: - (y - x + z)
+         */
+//        double LFpower = Range.clip(-(y+x+z),-1,1);
+//        double RFpower = Range.clip((y-x-z),-1,1);
+//        double LRpower = Range.clip(-(y-x+z),-1,1);
+//        double RRpower = Range.clip((y+x-z), -1, 1);
+
+        double forwrd = x; /* Invert stick X axis */
+        double strafe = y;
+
+        double pi = Math.PI;
+
+/* Adjust Joystick X/Y inputs by navX MXP yaw angle */
+
+        double gyro_radians = getYaw() * pi/180;
+        double temp = forwrd * cos(gyro_radians) +
+                strafe * sin(gyro_radians);
```

Comment:
#fix `double temp = forwrd * cos(gyro_radians) + strafe * sin(gyro_radians);` This calculation isn't so long as to need to be on multiple lines.